packages/flutter/lib/src/material/search_anchor.dart

```diff
@@ -1112,6 +1131,8 @@ class SearchController extends TextEditingController {
   // This is set automatically when a [SearchController] is given to the anchor
   // it controls.
   _SearchAnchorState? _anchor;
+  bool _usedByRoute = false;
+  bool _disposeByRoute = false;
```

Comment:
```suggestion
  bool _disposedByRoute = false;
```

Just to be consistent.